XenAdmin/TabPages/PhysicalStoragePage.cs

```diff
@@ -169,7 +172,17 @@ private void sr_PropertyChanged(object sender, PropertyChangedEventArgs e)
             }
             else
             {
-                Program.Invoke(this, () => RefreshRowForSr((SR)sender));
+                Program.Invoke(Program.MainWindow, RefreshAllItems);

```

Comment:
Why do you need to refresh all items instead of only the relevant row as the original code did?

Reply:
Because other tabpages also refresh like this.

Reply:
Ok, but it does not mean it's the right thing to do; in fact, if it may be a concern for performance if you have many entries in the gridview.

Reply:
I will refine this to only refresh particular rows.